streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -718,14 +719,20 @@ public void run() {
         try {
             runLoop();
             cleanRun = true;
+        } catch (final StreamsException e) {
```

Comment:
Yeah that's a good question.. My original plan is that we should not expect KafkaException to be thrown here since we already wrap all of them as StreamsException so if there's any thrown it may be a bug. But since either case we re-throw it anyways now I'm not feeling so strong (I've also thinking if we just do not re-throw exception if StreamsException since it is expected, but that is a behavior change since user's registered handler would not trigger then). I can just collapse all of them into one capture and just log the actual exception as well, wdyt?